src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -148,101 +159,180 @@ loadVTKMesh( Path const & filePath )
   return loadedMesh;
 }
 
-/**
- * @brief Compute assignments of mesh partitions produced by vtkRedistributeDataSetFilter to ranks.
- * @param numParts number of partitions (cuts)
- * @param numRanks number of MPI ranks
- * @return a vector of rank assignments
- *
- * This function merges partitions starting from the back, until the desired number is reached.
- * It is a stripped down version of vtkDIYKdTreeUtilities::ComputeAssignments(). We cannot call
- * that function directly because it's part of a VTK private module (i.e. header not installed).
- * We rely on copying the assignment procedure for lack of a better solution.
- * In particular, there is no API to extract this information out of vtkRedistributeDataSetFilter.
- */
-std::vector< int > computePartitionAssignments( int const numParts, int const numRanks )
+template< typename INDEX_TYPE >
+ArrayOfArrays< INDEX_TYPE, INDEX_TYPE >
+buildElemToNodes( vtkUnstructuredGrid & grid )
 {
-  std::vector< int > assignments;
-  assignments.reserve( numParts );
-  std::div_t const d = std::div( numParts, numRanks );
+  localIndex const numCells = LvArray::integerConversion< localIndex >( grid.GetNumberOfCells() );
+  array1d< INDEX_TYPE > nodeCounts( numCells );
 
-  for( int i = 0; i < numRanks; ++i )
+  // GetCell() is not thread-safe, use serial policy
+  forAll< serialPolicy >( numCells, [nodeCounts = nodeCounts.toView(), &grid] ( localIndex const cellIdx )
```

Comment:
If you're using no parallel feature, maybe doing simple `for` loops is OK?

Reply:
I'm actually refactoring this piece a little now that I've taken a second look. I found that VTK has a runtime indicator function that tells whether certain APIs of interest are thread-safe or not (I know, it's rather weird that it's runtime and not compile-time knowledge, but they have an option for switching internal storage type for connectivity arrays at runtime (!)). So I will be dispatching with either a serial or a parallel policy based on that, so the loop will stay as a RAJA kernel, but with POLICY passed as a template parameter.